source/AI.cpp

```diff
@@ -119,10 +119,74 @@ namespace {
 	void Deploy(const Ship &ship, bool includingDamaged)
 	{
 		for(const Ship::Bay &bay : ship.Bays())
-			if(bay.ship && (includingDamaged || bay.ship->Health() > .75))
+			if(bay.ship && (includingDamaged || bay.ship->Health() > .75) &&
+					(!bay.ship->IsYours() || bay.ship->IsDeploying()))
 				bay.ship->SetCommands(Command::DEPLOY);
 	}
 	
+	// Issue deploy orders for the selected ships (or the full fleet if no ships are selected).
+	void IssueDeploy(const PlayerInfo &player)
+	{
+		// Figure out what carried ships we are giving orders to
+		vector<Ship *> ships;
+		bool fullFleet = player.SelectedShips().empty();
+		if(fullFleet)
+		{
+			for(const shared_ptr<Ship> &it : player.Ships())
+				if(it.get() != player.Flagship() && !it->IsParked() && it->CanBeCarried())
+					ships.push_back(it.get());
+		}
+		else
+			for(const weak_ptr<Ship> &it : player.SelectedShips())
+			{
+				shared_ptr<Ship> ship = it.lock();
+				if(ship)
+					ships.push_back(ship.get());
+			}
+		
+		// This should never happen, but just in case:
+		if(ships.empty())
+		{
+			if(fullFleet)
+				Messages::Add("No carried ships in the fleet to deploy.");
```

Comment:
This message can show up a lot, if the player has 2 or more ships and no fighters or drones to deploy. Past behavior was that the deploy/recall message was only shown if you had carried ships in your active fleet.

Reply:
It now only shows up when the player hits the deploy button, and only when the player has no deployable ships in the fleet.
Do you prefer the deploy command to just remain silent in this situation? I have no strong feelings about this, but my first impression was that this message is better than just silently ignoring the command.